internal/util/parse/parse.go

```diff
@@ -84,11 +90,32 @@ func GitParseArgs(ctx context.Context, args []string) (Target, error) {
 }
 
 // targetFromPkgURL parses a pkg url and destination into kptfile git info and local destination Target
-func targetFromPkgURL(ctx context.Context, pkgURL, dest string) (Target, error) {
+func targetFromPkgURL(ctx context.Context, pkgURL string, dest string) (Target, error) {
 	g := Target{}
-	var repo, dir, version string
-	parts := strings.Split(pkgURL, ".git")
-	repo = strings.TrimSuffix(parts[0], "/")
+	repo, dir, version, err := parseURL(ctx, pkgURL)
+	if err != nil {
+		return g, err
+	}
+	destination, err := getDest(dest, repo, dir)
+	if err != nil {
+		return g, err
+	}
+	g.Ref = version
+	g.Directory = path.Clean(dir)
+	g.Repo = repo
+	g.Destination = filepath.Clean(destination)
+	return g, nil
+}
+
+// parseURL parses a pkg url and returns the repo, directory, and version
+func parseURL(ctx context.Context, pkgURL string) (repo string, dir string, version string, err error) {
+	parsedUrl, err := url.Parse(pkgURL)
+	if err != nil {
+		return "", "", "", err
+	}
+	parts := strings.Split(parsedUrl.Path, ".git")
```

Comment:
this may yield incorrect results with inputs such as:

https://my-site.com/root.gitops/foo

Imo we need to check for .git($|/)

Reply:
done, thank you!